src/GameLogic.cpp

```diff
@@ -3,4 +3,47 @@
 /*
 GameLogic class that holds all of the game objects and detects collisions
 and updates game status
-*/
\ No newline at end of file
+*/
+
+GameLogic::GameLogic() {
+    
+}
+
+void GameLogic::init(int wWidth, int wHeight) {
+    width = wWidth;
+    height = wHeight;
+    bulletSpeed = 500;
+    gameState = 1;
+    
+}
+
+
+void GameLogic::update(float time) {
+    if (gameState == 1) {
+        //perform game logic  
+    }
+    
+}
+
+void GameLogic::softReset() {
+    //return player character to beginning of level
+}
+
+void GameLogic::reset() {
+    //restart level? or restart game?
```

Comment:
This reset will probably be for restarting the whole game when you start a new game (instead of loading one) as there will not be any reason to reset any individual level once the player has started playing it.